test/lib/utils.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const {Cc, Ci, components} = require("chrome");
+const {stack: Cs} = components;
+
+function do_get_file(path, allowNonexistent) {
+  try {
+    let lf = Cc["@mozilla.org/file/directory_service;1"]
+      .getService(Ci.nsIProperties)
+      .get("CurWorkD", Ci.nsILocalFile);
+
+    let bits = path.split("/");
+    for (let i = 0; i < bits.length; i++) {
+      if (bits[i]) {
+        if (bits[i] == "..")
+          lf = lf.parent;
+        else
+          lf.append(bits[i]);
+      }
+    }
+
+    if (!allowNonexistent && !lf.exists()) {
+      // Not using do_throw(): caller will continue.
+      let stack = Cs.caller;
+      console.error(`[${stack.name} : ${stack.lineNumber}] ${lf.path} does not exist`);
+    }
+
+    return lf;
+  }
+  catch (ex) {
+    do_throw(ex.toString(), Cs.caller);
+  }
+
+  return null;
+}
+
+function do_throw(error, stack) {
```

Comment:
We use eslint for this project. And probably for all other projects as well. Gecko is not on eslint, makes things easy.

The naming is such because this code has been ported over from xpcshell, for better or for worse.

Since this has been taken with a few modifications, I think it would be fine to make it our own.